src/components/DocumentView/CodeBlock/highlight.ts

```diff
@@ -27,16 +29,31 @@ type InlineIndexed = { inline: any; start: number; end: number };
 
 type PositionedToken = ThemedToken & { start: number; end: number };
 
+/**
+ * Due to a combination of memory limitations of Cloudflare workers and the memory
+ * cost of shiki, we need to set a limit on the number of blocks we can highlight
+ * in a single page.
+ */
+let blockCount = 0;
+const BLOCK_LIMIT = 50;
```

Comment:
it's not ready to merge yet, storing this in a global variable means it's shared across executions. So one request for a page with a lot of blocks could impact the block loading of another, unrelated page